File: threads-compo/Threads.java
public class Threads {

    static class HelloThread implements Runnable {
        public void run() { 
            System.out.println("Hello world");
            try {
                Thread.sleep(1000);
            } catch (Exception ex) { 
                ex.printStackTrace();
            }
        }
    }

    public static void main(String... args) throws Exception {

        long start = System.currentTimeMillis();

        for (int i = 0; i < 100000; i++) {
            Thread thread = new Thread(new HelloThread());
            thread.start();
        }

        long stop = System.currentTimeMillis();

        System.out.println("Executed in: " + (stop - start) + "ms");

    }

}

File: threads-compo/threads.c
#include <stdio.h>
#include <pthread.h>
#include <time.h>

void* say_hello(void* arg) {
  printf("Hello world!\n");
  return NULL;
}

int main() {
  time_t start = time(NULL);
  int i;
  for (i = 0; i < 100000; i++) {
    pthread_t thread;
    void* status;
    pthread_create(&thread, NULL, say_hello, NULL);
    pthread_join(thread, &status);
  }
  time_t end = time(NULL);
  printf("Executed in: %d sec", end - start);
  return 0;
}
